### PR TITLE
refactor: change the position of signature hash in interoperation verify

### PR DESCRIPTION
**Pull request by driftluo**
**What this PR does / why we need it**:

This PR fix always success cell init

**Which issue(s) this PR fixes**:

Fixes #

**Which docs this PR relation**:

Ref #

**Which toolchain this PR adaption**:

No Breaking Change

**Special notes for your reviewer**:

NIL

**CI Description**

| CI Name                                   | Description                                                     |
| ----------------------------------------- | --------------------------------------------------------------- |
| *Chaos CI*                                | Test the liveness and robustness of Axon under terrible network condition    |
| *Cargo Clippy*                            | Run `cargo clippy --all --all-targets --all-features`      |
| *Coverage Test*                           | Get the unit test coverage report                             |
| *E2E Test*                                | Run end-to-end test to check interfaces                         |
| *Code Format*                             | Run `cargo +nightly fmt --all -- --check` and `cargo sort -gwc`     |
| *Web3 Compatible Test*                    | Test the Web3 compatibility of Axon                               |
| *v3 Core Test*                            | Run the compatibility tests provided by Uniswap V3             |
| *OCT 1-5 \| 6-10 \| 11 \| 12-15 \| 16-19* | Run the compatibility tests provided by OpenZeppelin           |

**CI Usage**

> Check the CI you want to run below, and then comment `/run-ci`.

**CI Switch**

- [ ] Chaos CI
- [x] Cargo Clippy
- [ ] Coverage Test
- [ ] E2E Tests
- [x] Code Format
- [x] Unit Tests
- [x] Web3 Compatible Tests
- [ ] OCT 1-5 And 12-15
- [ ] OCT 6-10
- [ ] OCT 11
- [ ] OCT 16-19
- [ ] v3 Core Tests

### CODE REVIEW

**Review by KaoImin — DISMISSED** [review]

**Comment by KaoImin** [comment]
hold

File: protocol/src/traits/interoperation.rs [hunk 1]
```diff
@@ -2,29 +2,26 @@ use ckb_traits::{CellDataProvider, HeaderProvider};
...
-/// capacity: 0x277cf2a00  
+/// data: signature hash(32 bytes)
+/// capacity: 0x1ad274800
 /// ```
-/// So the occupied bytes is 32 + 32 + 1 + 32 + 1 + 8 = 106 bytes.
-const fn always_success_cell_bytes() -> u64 {
-    32 + 32 + 1 + 32 + 1 + 8
+/// So the occupied bytes is 32 + 32 + 8 = 72 bytes.
+const fn signature_hash_cell_bytes() -> u64 {
+    32 + 32 + 8
```
Comment (Flouse) [inline protocol/src/traits/interoperation.rs]: Is it missing lock.hash_type's one byte?

Comment (chaoticlonghair) [reply protocol/src/traits/interoperation.rs]: I suggest add a unit test that call [`packed::CellOutput::occupied_capacity(..)`](https://github.com/nervosnetwork/ckb/blob/290eeed780eaf3f67beef7c8820187714de3c339/util/types/src/extension/capacity.rs#L14-L16) with a real cell.

```rust
// I did not test the follow code.
#[test]
fn test_const_signature_hash_cell_bytes() {
    let excepted = signature_hash_cell_bytes();
    let data_capacity = Capacity::bytes(32);
    let actual = packed::CellOutputBuilder::default()
        .build()
        .occupied_capacity(data_capacity)
        .unwrap()
        .shannons();
    assert_eq!(excepted, actual);
}
```

Or [`packed::CellOutputBuilder::build_exact_capacity(..)`](https://github.com/nervosnetwork/ckb/blob/0a3f9b16a1bd2b928b6c82872cfa98ed7b3accc3/util/types/src/extension/capacity.rs#L59).

```rust
// I did not test the follow code, too.
#[test]
fn test_const_signature_hash_cell_bytes() {
    let excepted = signature_hash_cell_bytes();
    let data_capacity = Capacity::bytes(32);
    let actual = packed::CellOutputBuilder::default()
        .build()
        .build_exact_capacity(data_capacity)
        .unwrap()
        .capacity()
        .shannons();
    assert_eq!(excepted, actual);
}
```

**Review by Flouse — APPROVED** [review]
LGTM

**Review by chaoticlonghair — APPROVED** [review]

**Pull request merged**